core/shared/src/main/scala/fs2/Stream.scala

```diff
@@ -3024,6 +3024,39 @@ object Stream extends StreamLowPriority {
         .eval(timer.clock.monotonic(NANOSECONDS).map(now => (now - start).nanos))
     }
 
+  def foldEvery[F[_], A, B](
+      d: FiniteDuration
+  )(init: B)(f: (B, A) => B)(implicit F: Sync[F]): Pipe[F, A, B] = { in =>
+    {
+      val width = d.toMillis
+
+      val signal = Stream.eval(F.delay(System.currentTimeMillis())).repeat.zipWithPrevious.map {
```

Comment:
This should use `Timer[F]` rather than `System`

Reply:
For sure, that would definitely be part of cleanup! I just thought I'd get some feedback on whether people would like it to be part of fs2, before I invest time in cleaning it up, writing tests, etc